feat: store multiple distributor types per address in array format

yahgw-ai opened the pull request:
## Summary
- Changed DistributorsData type to store DistributorInfo[] instead of single DistributorInfo
- Modified DistributorDetector to append multiple types for same address instead of silently discarding
- Updated all downstream components to use first element from array

## Test plan
- [x] Added unit tests for array validation in FileManager
- [x] Added test for multiple distributor types per address
- [x] Verified integration test shows both types are stored
- [ ] Fix remaining test files to use array format
- [ ] Run full test suite
- [ ] Update documentation

Closes #277

yahgw-ai committed:
feat: implement array structure for distributor data

- Updated DistributorsData type to store DistributorInfo[] instead of single DistributorInfo
- Modified FileManager validation to handle and validate array structure
- Updated DistributorDetector to append multiple types for same address
- Modified downstream components to use first element from array
- Added tests for multiple distributor types per address
- Fixed undefined checks in downstream components

BREAKING CHANGE: distributors are now stored as arrays per address

yahgw-ai committed:
fix: update tests to match DistributorsData array structure

Updated all test files to properly use arrays for distributor info instead of single objects.
The DistributorsData type expects distributors to be { [address: string]: DistributorInfo[] }.

yahgwai commented on src/core/fee-calculation/recipient-recieved-scanner.ts:
```diff
@@ -131,13 +131,22 @@ export class RecipientRecievedScanner {
+      const distributorInfo = distributorInfoArray[0];
+
+      if (!distributorInfo) {
+        console.log(
```
this should throw an error - since we already checked that the array is non empty it should only contain valid distributors

yahgw-ai replied:
Fixed - Changed to throw an error instead of logging and continuing. The error now clearly indicates invalid distributor data.

yahgwai commented on src/core/fee-calculation/balance-fetcher.ts:
```diff
@@ -121,12 +121,21 @@ export class BalanceFetcher {
+      const distributorInfo = distributorInfoArray[0];
+
+      if (!distributorInfo) {
+        console.log(
```
Again, should throw an error

yahgw-ai replied:
Fixed - Changed to throw an error here as well, maintaining consistency with the recipient-received-scanner change.

yahgwai commented on scripts/migrate-distributor-tests.js:
```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env node
```
We dont need to keep this script

yahgw-ai replied:
Removed - deleted scripts/migrate-distributor-tests.js

yahgwai commented on scripts/fix-distributor-tests.js:
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env node
```
Also dont need to keep this script

yahgw-ai replied:
Removed - deleted scripts/fix-distributor-tests.js

yahgw-ai committed:
fix: address PR review comments

- Change error handling to throw errors instead of logging and continuing when
  distributor array contains falsy elements (balance-fetcher and recipient-recieved-scanner)
- Remove unnecessary migration scripts (migrate-distributor-tests.js and fix-distributor-tests.js)
